### PR TITLE
Use Java Process API to launch Axon Ivy Engine

### PR DESCRIPTION
**Pull request by alexsuter**
- No longer need commons.exec
- No need to SIGKILL engine in tests which occurs in side-effects on windows and ugly errors in build log

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -58,9 +58,7 @@ def collectBuildArtifacts()  {
   junit testDataPublishers: [[$class: 'AttachmentPublisher'], [$class: 'StabilityTestDataPublisher']], testResults: '**/target/surefire-reports/**/*.xml'
   recordIssues tools: [mavenConsole()], qualityGates: [[threshold: 1, type: 'TOTAL']], filters: [
     excludeType('site-maven-plugin:site'),
-    excludeType('maven-surefire-plugin:test'),
-    // printed to console by test. was since ever the case but they are now real maven warnings
-    excludeMessage('.*Uncaught exception in thread Thread.*'),
```
Comment (alexsuter) [inline Jenkinsfile:63]: no longer a problem

File: build/windows/Jenkinsfile [hunk 2]
```diff
@@ -39,13 +39,10 @@ pipeline {
 def collectBuildArtifacts()  {
   archiveArtifacts 'target/*.jar'
   archiveArtifacts 'target/its/**/build.log'
-  // skipMarkingBuildUnstable: true for the mean time until tests are fixed
```
Comment (alexsuter) [inline build/windows/Jenkinsfile:42]: enable tracking of junit and maven warnings for windows build

File: src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java [hunk 3]
```diff
@@ -214,9 +207,9 @@ private void assertFileExecutable(Path file) {
             .isExecutable();
   }
 
-  private static void kill(Executor startedProcess) {
+  private static void kill(Process startedProcess) {
     if (startedProcess != null) {
-      startedProcess.getWatchdog().destroyProcess();
+      startedProcess.destroy();
```
Comment (alexsuter) [inline src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java:212]: will make a sigterm and longer a sigkill

File: src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java [hunk 4]
```diff
@@ -57,16 +55,11 @@ public void canStartEngine() throws Exception {
   }
 
   @Test
-  public void testKillEngineOnVmExit() throws Exception {
+  public void killEngineOnVmExit() throws Exception {
```
Comment (alexsuter) [inline src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java:58]: TODO

File: src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java [hunk 5]
```diff
@@ -179,8 +162,9 @@ public void startEngine_targetDirectoryNotClean() throws Exception {
       startedProcess = mojo.startEngine();
       assertThat(engineDirTarget).exists();
       assertThat(log.getWarnings().toString()).doesNotContain("Skipping copy");
-
       kill(startedProcess);
+
+      //Thread.sleep(5000);
```
Comment (ivy-rew) [inline src/test/java/ch/ivyteam/ivy/maven/test/TestStartEngine.java:167]: :rabbit:

**Comment by ivy-rew** [comment]
sounds great :guitar:

**Review by ivy-rew — APPROVED** [review]

File: src/main/java/ch/ivyteam/ivy/maven/engine/EngineControl.java [hunk 6]
```diff
@@ -133,7 +134,10 @@ private ProcessBuilder toProcessBuilder(Command command) {
     cmds.add("-Dosgi.install.area=" + osgiDir.toAbsolutePath());
 
     if (StringUtils.isNotBlank(context.vmOptions.additionalVmOptions())) {
-      cmds.add(context.vmOptions.additionalVmOptions()); // todo no escape
```
Comment (alexsuter) [inline src/main/java/ch/ivyteam/ivy/maven/engine/EngineControl.java]: Hello @ivy-rew 

I've one issue here. The additionalVmOptions were passed unescaped to commons.exec as one-liner. This is not possible with Java Process API.

A quick-dirty hack which might not work for all uses cases is to split them by whitespace, but I think the real solution would be to change additionalVmOptions from a one-liner to a List.

You see such a pattern also in other plugins:
https://maven.apache.org/plugins/maven-compiler-plugin/compile-mojo.html#compilerArgs

What do you think should we go with this solution, which works at least for engine-integration-test or should I already introduce a new one and deprecate the old one? And implement the old one with this approach?

Comment (ivy-rew) [reply src/main/java/ch/ivyteam/ivy/maven/engine/EngineControl.java]: Yes, I would definitively support a real list argument type, just like we also do to configure JDT: https://axonivy.github.io/project-build-plugin/release/11.3/compileProject-mojo.html#compilerOptions

And another yes, I'd deprecate the old parameter and make it work by analysing whitespaces ... or maybe glean some code from commons.exec that did this logic. Though I do not think it's worth to investigate a lot into this backwoard compatibility ... I think these parameters have rarely been configured in projects.

Comment (alexsuter) [reply src/main/java/ch/ivyteam/ivy/maven/engine/EngineControl.java]: Sounds good. I will make a new PR for this.

**Comment by alexsuter** [comment]
fyi @ivy-cst

**Pull request merged**